CenterForOpenScience/osf.io
# [ENG-2543][ENG-2642] clean up how metadata is pushed to SHARE

## Purpose
SHARE will be changing soon, and will not be merging data from different
sources together -- let's stop relying on that behavior


## Changes
- update the metadata record sent to share:
  - include the full node lineage (parent, grandparent, etc. up to the
  root) for registrations and projects, including metadata (title and
  guid-url identifier) for each – building the lineage for search
  results previously depended on merging multiple records together

  - include full metadata for non-registration nodes – previously
  only marked a project deleted, relying on merging with harvested
  data to fill in the rest

  - don’t include emails or "profile image url" as identifiers

  - don't include non-bibliographic contributors

  - add `suid` and `central_node_id` to the pushed payload – makes it
  unambiguous what object the metadata record describes, so SHARE
  doesn’t have to guess
  - use `is_spam` instead of `is_spammy` -- only delete from share if *confirmed* spam
- add `recatalog_metadata` management command, with chunking and resumability
  - e.g. `python3 manage.py recatalog_metadata --preprints --all-providers`


## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify updates to projects, preprints, and registrations are correctly reflected in SHARE-based search results
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Documentation



## Side Effects



## Ticket
[ENG-2543]
[ENG-2642]



[ENG-2543]: https://openscience.atlassian.net/browse/ENG-2543
